Label Transform (#1708)
vega/vega: docs/docs/transforms/label.md

---
layout: transform
title: Label Transform
permalink: /docs/transforms/label/index.html
---

> The **label** transform repositions texts in text mark, so that their placements are not colliding with other elements in the chart. Those elements include marks that have name included in property `avoidMarks` and the mark that is used as backing data (as explained in [reactive geometry](../../marks/#reactivegeom)) of the text mark this label transform is transforming.

Reviewer: ```suggestion
The **label** transform repositions text marks so that they are not colliding with other elements in the chart. The label transform avoids all marks of types listed in `avoidMarks` and the mark that is used as backing data (as explained in [reactive geometry](../../marks/#reactivegeom)) of the text mark this label transform is using as input.
```

> The label transform is useful for labeling data points by creating a text mark that takes in data from the mark that represents the data point (we will call this 'base mark' for this label documentation), then use the label transform on the text mark to reposition the text, so that the texts appear near their data points without colliding into other objects in the chart.

Reviewer: ```suggestion
The label transform is useful for labeling data points by creating a text mark that takes in data from the mark that represents the data point (we will call this mark the "base mark"). You can then use the label transform on the text mark to reposition the text so that the texts appear near their data points without colliding with other objects in the chart.
```

| markIndex     |  {% include type t="Number" %}  | `markIndex` is used to specify which mark in the group mark you want to label because group mark can have more than 1 mark in the group. Regularly, `markIndex` is used with group-line and group-area mark. **Note**: `markIndex` only works with group mark as the base mark. **Default value:** `0`                                                                                                                     |
| as            | {% include type t="String[]" %} | The output fields written by the transform. The default is `['x', 'y', 'opacity', 'align', 'baseline', 'originalOpacity', 'transformed']`                                                                                                                                                                                                                                                                                  |

## Usage

### <a name="concept"></a>Basic concept

{% include embed spec="label-examples/basic-concept-no-label" %}

> In this scatter plot example, we can add label to each point by adding text mark with the same `x` and `y` encoding channels to the chart. Since `x` and `y` encoding channels of the text mark follow the ones from the symbol mark (as we call this base mark), we can use reactive geometry to get information from the symbol mark and only modify the text.

Reviewer: ```suggestion
In this scatter plot example, we can add labels to each point by adding a text mark with the same `x` and `y` encoding channels to the chart. Since `x` and `y` encoding channels of the text mark follow the ones from the symbol mark (as we call this base mark), we can use reactive geometry to get information from the symbol mark and only modify the text.
```
